### ISSUE TITLE
Question about callback order and connection lifecycle

### ISSUE THREAD

**Issue by nixomose** [warmcat/libwebsockets#1059]
When I open one lws connection everything is fine, when I open two for the same protocol weird things start happening. Likely on my side, I just wanted to verify a few things.

If I allocate the user pointer memory and pass it in the connection info, when it is safe to free it?
At first I thought it was LWS_CALLBACK_CLOSED, but it seems I can still get LWS_CALLBACK_CLIENT_WRITEABLE after that. So then I found LWS_CALLBACK_WSI_DESTROY. But that is only called on protocol zero, I thought wsi was a pointer per connection (the lws *).

It is possible I'm getting client_writable callbacks because another thread is generating them via lws_callback_on_writable() and that thread hasn't noticed that the connection closed yet, so that might be me.

So in short my question is: when is it safe to free the user pointer memory I allocated at the start of the connection?

Thanks.

**Comment by nixomose** [comment]
I think I rescind my question.
I am definitely getting a write callback after I get a close, but it's being triggered by another thread. Once I fixed the race condition I could see it more clearly.
So now I am freeing the user pointer memory on the close call and ignoring any write calls backs after that and I guess that should work.
